service/src/main/java/bio/terra/cbas/dao/MethodDao.java

```diff
@@ -95,4 +95,28 @@ public int updateLastRunWithRunSet(RunSet runSet) {
                 "run_set_id", runSet.runSetId(),
                 "method_id", runSet.methodVersion().method().methodId())));
   }
+
+  public int getNumberOfMethods(
+      String methodName, String methodUrl, String methodVersion, String methodSource) {
+
+    String sql =
+        "SELECT COUNT(*) FROM method INNER JOIN method_version "
+            + "ON method.method_id = method_version.method_id "
+            + "WHERE method.name = :name "
+            + "AND method_version.method_version_url = :methodVersionUrl "
+            + "AND method_version.method_version_name = :methodVersionName "
+            + "AND method.method_source = :methodSource";
+
+    MapSqlParameterSource params =
+        new MapSqlParameterSource(
+            Map.of(
+                "name", methodName,
+                "methodVersionUrl", methodUrl,
+                "methodVersionName", methodVersion,
+                "methodSource", methodSource));
+
+    Integer count = jdbcTemplate.queryForObject(sql, params, Integer.class);
+
+    return count;
```

Comment:
You could probably do this all in one line `return jdbcTemplate.queryForObject...`